src/main/java/com/jcraft/jsch/Session.java

```diff
@@ -2140,6 +2193,13 @@ public void disconnect() {
     }
     io = null;
     socket = null;
+    l.lock();
+    try {
+      disconnectingChannels = false;
+      channels.clear();
+    } finally {
+      l.unlock();
+    }
```

Comment:
I added this near the end of `disconnect()` after all the IO operations finished and threads stopped, so that the session returns to a blank slate, as users in the past have expressed a desire to reuse session objects.
Without this, the `SessionReconnectIT` would end up failing, since it tests the reuse operation.